src/Blocks/BlockConcretePowder.h

```diff
@@ -0,0 +1,98 @@
+
+#pragma once
+
+#include "BlockHandler.h"
+
+
+
+
+
+class cBlockConcretePowderHandler :
+	public cBlockHandler
+{
+public:
+	cBlockConcretePowderHandler(BLOCKTYPE a_BlockType)
+		: cBlockHandler(a_BlockType)
+	{
+	}
+
+
+
+
+
+	virtual void Check(cChunkInterface & a_ChunkInterface, cBlockPluginInterface & a_PluginInterface, int a_RelX, int a_RelY, int a_RelZ, cChunk & a_Chunk) override
+	{
+		GetSoaked(Vector3i(a_RelX, a_RelY, a_RelZ), a_Chunk);
+	}
+
+
+
+
+
+	/** Check blocks above and around to see if they are water. If one is, convert this into concrete block */
+	void GetSoaked(Vector3i a_Rel, cChunk & a_Chunk)
+	{
+		bool ShouldSoak = false;
+
+		static const Vector3i Coords[] =
+		{
+			Vector3i( 1, 0,  0),
+			Vector3i(-1, 0,  0),
+			Vector3i( 0, 0,  1),
+			Vector3i( 0, 0, -1),
+			Vector3i( 0, 1,  0),
+		};
+		for (size_t i = 0; i < ARRAYCOUNT(Coords); i++)
```

Comment:
This loop can be replaced with `std::any_of`.

Reply:
Has this been used before in Cuberite?

Reply:
Admittedly, my search only finds one:
https://github.com/cuberite/cuberite/blob/6309c6a97fdbabfde978358f5f9a0f61ab74f91f/src/Simulator/FireSimulator.cpp#L124-L129
and it was only added yesterday at my request.

However, using an algorithms gives the loop a name.  That makes it much better at expressing the real intent: "a concrete block should soak if _any of_ its neighbours are water".  It's the exact same reason why `std::min(a, b)` is preferred over `(a < b) ? a : b`.

Reply:
Okay. Implemented.